gradle-guide-error-prone/src/test/java/com/palantir/gradle/guide/errorprone/ConfigurationAvoidanceRegistrationTest.java

```diff
@@ -0,0 +1,292 @@
+                    tasks.create("lol", Task.class, task -> {});
+                }
+
+                static Task used_return_value(TaskContainer tasks) {
+                    // BUG: Diagnostic contains: use `.register`
+                    Task task = tasks.create("lol");
+                    // BUG: Diagnostic contains: use `.register`
+                    System.out.println(tasks.create("lol", Task.class, t -> {}));
+                    // BUG: Diagnostic contains: use `.register`
+                    return tasks.create("lol", Task.class);
+                }
+
+                static void methods_with_no_directly_equivalent_register(TaskContainer tasks) {
+                    // BUG: Diagnostic contains: use `.register`
+                    tasks.create("lol", Closure.IDENTITY);
+                    // BUG: Diagnostic contains: use `.register`
+                    tasks.create(Map.of("name", "lol", "type", Task.class));
+                    // BUG: Diagnostic contains: use `.register`
+                    tasks.create(Map.of("name", "lol", "type", Task.class), Closure.IDENTITY);
+                }
+            }
+            """)
+                .addOutputLines(
+                        "Test.java",
+                        // language=java
+                        """
+            import groovy.lang.Closure;
+            import java.util.Map;
+            import org.gradle.api.Task;
+            import org.gradle.api.tasks.TaskContainer;
+            import org.gradle.api.tasks.TaskProvider;
+
+            class Test {
+                static void unused_return_value(TaskContainer tasks) {
+                    tasks.register("lol");
+                    tasks.register("lol", Task.class);
+                    tasks.register("lol", Task.class, new Object());
+                    tasks.register("lol", Task.class, task -> {});
+                }
+
+                static Task used_return_value(TaskContainer tasks) {
+                    TaskProvider<Task> task = tasks.register("lol");
+                    System.out.println(tasks.register("lol", Task.class, t -> {}).get());
+                    return tasks.register("lol", Task.class).get();
+                }
+
+                static void methods_with_no_directly_equivalent_register(TaskContainer tasks) {
+                    // BUG: Diagnostic contains: use `.register`
+                    tasks.create("lol", Closure.IDENTITY);
+                    // BUG: Diagnostic contains: use `.register`
+                    tasks.create(Map.of("name", "lol", "type", Task.class));
+                    // BUG: Diagnostic contains: use `.register`
+                    tasks.create(Map.of("name", "lol", "type", Task.class), Closure.IDENTITY);
+                }
+            }
+            """)
+                .doTest();
+    }
+
+    @Test
+    void best_effort_refactors_usages_of_configurations_create_where_the_return_value_is_unused_to_register() {
```

Comment:
separate calls pls 🙏  I'm having to diff bigger chunks + scrolling => not fun

Reply:
Done